wordpress-mobile/WordPress-Android
# [Gutenberg] Add `metadata` parameter when inserting media into editor

**Related PRs:**
* https://github.com/WordPress/gutenberg/pull/48994
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5558

Pass the VideoPress ID when inserting media from the Media library.

**To test:**

The `metadata` parameter is not used yet, so we need to check if the value is logged in the console. In this case, we'll test the `videopressGUID` metadata parameter passed on video items.

1. Build the app locally.
2. Connect the app with the local React Native Metro server.
3. Apply the following patch file:
```patch
diff --git forkSrcPrefix/packages/block-library/src/video/edit.native.js forkDstPrefix/packages/block-library/src/video/edit.native.js
index 99225ba5d5c569d19a158f5b1a203c68aa353c33..194c116ae81ad01c7430fc5a3dbd3bb9878f490b 100644
--- forkSrcPrefix/packages/block-library/src/video/edit.native.js
+++ forkDstPrefix/packages/block-library/src/video/edit.native.js
@@ -154,9 +154,10 @@ class VideoEdit extends Component {
 		this.setState( { isUploadInProgress: false } );
 	}
 
-	onSelectMediaUploadOption( { id, url } ) {
+	onSelectMediaUploadOption( { id, url, metadata } ) {
 		const { setAttributes } = this.props;
 		setAttributes( { id, src: url } );
+		console.log( { metadata } );
 	}
 
 	onSelectURL( url ) {
```
4. Go to a Simple site (videos uploaded in Simple sites are added automatically to VideoPress).
5. Go to the Media screen.
6. Upload a video.
7. Open/create a post.
8. Add a Video block.
9. Select "WordPress Media Library" option.
10. Select the uploaded video.
11. Check the logs and observe that the metadata object is logged with the VideoPress GUID. I.e. `{"metadata": {"videopressGUID": "<VIDEOPRESS_GUID"}}`

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
